source/Calamari/Kubernetes/Commands/Executors/GatherAndApplyRawYamlExecutor.cs

```diff
@@ -43,12 +41,16 @@ protected override async Task<IEnumerable<ResourceIdentifier>> ApplyAndGetResour
             
             if (globs.IsNullOrEmpty())
                 return Enumerable.Empty<ResourceIdentifier>();
-            
-            var directories = GroupFilesIntoDirectories(deployment, globs, variables);
-            
+
+            var globberGrouping = new GlobberGrouping(fileSystem);
+            var globDirectories = globberGrouping.Group(deployment.CurrentDirectory, globs);
+            variables.Set(SpecialVariables.GroupedYamlDirectories, string.Join(";", globDirectories.Select(d => d.Directory)));
+
             var resourcesIdentifiers = new HashSet<ResourceIdentifier>();
-            foreach (var directory in directories)
+            for (int i = 0; i < globDirectories.Count(); i++)
```

Comment:
We seem to use the index number for the message below. Rather than putting this logging specific property on the GlobDirectory object, it has been changed to use this method.